lib/src/client.dart

```diff
@@ -37,15 +39,25 @@ class Client {
     _cache = cache;
   }
 
+  set apiTokenPrefix(String value) {
+    _apiTokenPrefix = value;
+  }
+
   // Getters
   String get endPoint => this._endpoint;
 
   String get apiToken => this._apiToken;
 
-  Map<String, String> get headers => {
-        'Authorization': 'Bearer $apiToken',
-        'Content-Type': 'application/json',
-      };
+  String get apiTokenPrefix => this._apiTokenPrefix;
+
+  Map<String, String> get headers {
+    Map<String, String> h = new Map();
+    if (apiToken != null) {
+      h['Authorization'] = '$apiTokenPrefix $apiToken';
```

Comment:
Now you are fixing the issue where the user has a different `apiTokenPrefix`, but what happens when the `Authorization ` has a different format?

Maybe allow the user to define the getter function for Authorization header, therefore removing the need for `apiTokenPrefix ` when the user defines his own function, or uses `'Bearer'` by default.

Reply:
That middleware adds unnecessary overhead to every request.